conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DialogueClientGenerator.java

```diff
@@ -0,0 +1,586 @@
+
+    private static final Class<?> EMPTY_BODY_CLASS = EmptyBody.class;
+    private static final TypeName EMPTY_BODY = TypeName.get(EMPTY_BODY_CLASS);
+
+    @Override
+    public Set<JavaFile> generate(ConjureDefinition conjureDefinition) {
+        TypeMapper parameterTypes = new TypeMapper(conjureDefinition.getTypes(),
+                new ClassVisitor(conjureDefinition.getTypes(), ClassVisitor.Mode.PARAMETER));
+        TypeMapper returnTypes = new TypeMapper(conjureDefinition.getTypes(),
+                new ClassVisitor(conjureDefinition.getTypes(), ClassVisitor.Mode.RETURN_VALUE));
+        TypeAwareGenerator generator = new TypeAwareGenerator(parameterTypes, returnTypes);
+        return conjureDefinition.getServices().stream()
+                .flatMap(serviceDef -> generator.service(serviceDef).stream())
+                .collect(Collectors.toSet());
+    }
+
+    // TODO(rfink): Split into separate classes: endpoint, interface, impl.
+    private static final class TypeAwareGenerator {
+        private static final String AUTH_HEADER_NAME = "Authorization";
+        private static final String AUTH_HEADER_PARAM_NAME = "authHeader";
+        private static final String MAPPER_FIELD_NAME = "mapper";
+        private static final String REQUEST_VAR_NAME = "_request";
+        private static final String REQUEST_VAR_NAME_BUILDER = "_request_builder";
+
+        private final TypeMapper parameterTypes;
+        private final TypeMapper returnTypes;
+
+        private TypeAwareGenerator(TypeMapper parameterTypes, TypeMapper returnTypes) {
+            this.parameterTypes = parameterTypes;
+            this.returnTypes = returnTypes;
+        }
+
+        private Set<JavaFile> service(ServiceDefinition def) {
+            return ImmutableSet.of(blockingApi(def), client(def));
+        }
+
+        private JavaFile blockingApi(ServiceDefinition def) {
+            TypeSpec.Builder serviceBuilder = TypeSpec.interfaceBuilder(serviceClassName("Blocking", def))
+                    .addModifiers(Modifier.PUBLIC)
+                    .addAnnotation(ConjureAnnotations.getConjureGeneratedAnnotation(DialogueClientGenerator.class));
+
+            def.getDocs().ifPresent(docs ->
+                    serviceBuilder.addJavadoc("$L", StringUtils.appendIfMissing(docs.get(), "\n")));
+
+            serviceBuilder.addMethods(def.getEndpoints().stream()
+                    .map(this::blockingApiMethod)
+                    .collect(Collectors.toList()));
+
+            return JavaFile
+                    .builder(def.getServiceName().getPackage(), serviceBuilder.build())
+                    .skipJavaLangImports(true)
+                    .build();
+        }
+
+        private MethodSpec blockingApiMethod(EndpointDefinition endpointDef) {
+            MethodSpec.Builder methodBuilder = MethodSpec.methodBuilder(endpointDef.getEndpointName().get())
+                    .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
```

Comment:
curious: are `PUBLIC` `ABSTRACT` necessary since it's the "method" of an interface (and obv javapoet knows this since it doesn't render those modifiers). What happens if you remove those modifiers ?

Reply:
`java.lang.IllegalArgumentException: modifiers [abstract] must contain one of [public, private]`

Reply:
Found that: https://github.com/square/javapoet#interfaces

Makes sense. Thanks!